deploy/olm-catalog/kubevirt-hyperconverged/1.2.0/hco00.crd.yaml

```diff
@@ -150,6 +150,78 @@ spec:
+                            type: object
+                          tolerations:
+                            description: tolerations is a list of tolerations applied to the relevant kind of pods See https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/ for more info. These are additional tolerations other than default ones.
+                            items:
+                              description: The pod this Toleration is attached to tolerates any taint that matches the triple <key,value,effect> using the matching operator <operator>.
+                              properties:
+                                effect:
+                                  description: Effect indicates the taint effect to match. Empty means match all taint effects. When specified, allowed values are NoSchedule, PreferNoSchedule and NoExecute.
+                                  type: string
+                                key:
+                                  description: Key is the taint key that the toleration applies to. Empty means match all taint keys. If the key is empty, operator must be Exists; this combination means to match all values and all keys.
+                                  type: string
+                                operator:
+                                  description: Operator represents a key's relationship to the value. Valid operators are Exists and Equal. Defaults to Equal. Exists is equivalent to wildcard for value, so that a pod can tolerate all taints of a particular category.
+                                  type: string
+                                tolerationSeconds:
+                                  description: TolerationSeconds represents the period of time the toleration (which must be of effect NoExecute, otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint forever (do not evict). Zero and negative values will be treated as 0 (evict immediately) by the system.
+                                  format: int64
+                                  type: integer
+                                value:
+                                  description: Value is the taint value the toleration matches to. If the operator is Exists, the value should be empty, otherwise just a regular string.
+                                  type: string
+                              type: object
+                            type: array
+                        type: object
+                      workloads:
+                        description: workloads NodePlacement defines the NodePlacement (nodeSelector and tolerations) for all the workloads pods managed by operators managed by the HyperConverged Operator. Changes to Workloads NodePlacement can be applied only without existing workload.
```

Comment:
I think we can make this here more clear. Let's add something here like:

`influences the placement of intrastructure components which need to be running on a node where virtualization workloads should be able to run`.

Let me again add that CDI may not want to interpret this. I would certainly not expect that my image importer runs on master nodes just because I want to make it possible to run a VM on master nodes. On the other hand there are cases where this is probably necessary (like with local volume provisioning). A little bit tricky. We should make a clear statement here, so that everyone knows what to expect for CDI.

Reply:
> influences the placement of intrastructure components which need to be running on a node where virtualization workloads should be able to run.

sounds good to me. 

> Let me again add that CDI may not want to interpret this. I would certainly not expect that my image importer runs on master nodes just because I want to make it possible to run a VM on master nodes. On the other hand there are cases where this is probably necessary (like with local volume provisioning). A little bit tricky. We should make a clear statement here, so that everyone knows what to expect for CDI.

I agree CDI is tricky. I'd caution us from to tackle or directly address documenting that here at the HCO level... at least right now.

This is opening up a discussion that needs to be had at the CDI level, and it's one that we might not get 100% right the first go around. I'd let that evolve independently of the HCO. We at least agree on the high level categories of worker and infra placement. CDI can sort out what belongs where.

Reply:
I am under the impression we have waitforfirstconsumer for the purpose of scheduling the VM and the storage on the same node.
I imagine that if someone is using HCO, they see the sub-components as one project and expect the settings to be propagated to them.

Reply:
I'd prefer to avoid using `intrastructure components` terminology for workloads placement if we named the other category `infra`.

Reply:
Yes the WFFC code we are adding right is expressly to make sure we provision the storage on the nodes that kubevirt wants, instead of nodes that happen to be able to run CDI pods. So if CDI follows the same node selectors for workloads as kubevirt and waits for kubevirt to spawn a VM pod that binds the storage to that node, then CDI should always do the right thing.